core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -4283,6 +4283,157 @@ private void checkLiteral2(String expression, String expected) {
     });
   }
 
+  @Test
+  public void testTONUMBERFunctionHandlingHexaToInt() {
+    String query = "select TO_NUMBER('03ea02653f6938ba','XXXXXXXXXXXXXXXX')";
+    final String expected = "SELECT CAST(CONCAT('0x', '03ea02653f6938ba') AS INTEGER)";
+    sql(query).withBigQuery().ok(expected);
+  }
+
+  @Test
+  public void testTONUMBERFunctionHandlingFloatingPoint() {
+    String query = "select TO_NUMBER('1.789','9.999')";
+    final String expected = "SELECT CAST('1.789' AS FLOAT)";
+    sql(query).withBigQuery().ok(expected);
+    sql(query).withHive().ok(expected);
+    sql(query).withSpark().ok(expected);
+  }
+
+  @Test
+  public void testTONUMBERFunctionHandlingFloatingPointWithD() {
+    String query = "select TO_NUMBER('1.789','9D999')";
+    final String expected = "SELECT CAST('1.789' AS FLOAT)";
+    sql(query).withBigQuery().ok(expected);
+    sql(query).withHive().ok(expected);
+    sql(query).withSpark().ok(expected);
+  }
+
+  @Test
+  public void testTONUMBERFunctionHandlingWithComma() {
+    String query = "SELECT TO_NUMBER ('1,789', '9,999')";
+    final String expected = "SELECT CAST('1789' AS INTEGER)";
+    sql(query).withBigQuery().ok(expected);
+    sql(query).withHive().ok(expected);
+    sql(query).withSpark().ok(expected);
+  }
+
+  @Test
+  public void testTONUMBERFunctionHandlingWithCurrency() {
+    String query = "SELECT TO_NUMBER ('$1789', '$9999')";
+    final String expected = "SELECT CAST('1789' AS INTEGER)";
+    sql(query).withBigQuery().ok(expected);
+    sql(query).withHive().ok(expected);
+    sql(query).withSpark().ok(expected);
+  }
+
+  @Test
+  public void testTONUMBERFunctionHandlingWithCurrencyAndL() {
+    String query = "SELECT TO_NUMBER ('$1789', 'L9999')";
+    final String expected = "SELECT CAST('1789' AS INTEGER)";
+    sql(query).withBigQuery().ok(expected);
+    sql(query).withHive().ok(expected);
+    sql(query).withSpark().ok(expected);
```

Comment:
Follow calcite formatting for test cases.

Reply:
Done